.github/workflows/collection-continuous-integration.yml

```diff
@@ -0,0 +1,158 @@
+name: Collection test suite
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: 3 0 * * *  # Run daily at 0:03 UTC
+
+jobs:
+  build-collection-artifact:
+    name: Build collection
+    runs-on: ${{ matrix.runner-os }}
+    strategy:
+      matrix:
+        runner-os:
+          - ubuntu-latest
+          - ubuntu-16.04
+        ansible-version:
+          - ansible==2.9.5
+          - git+https://github.com/ansible/ansible.git@devel
+        runner-python-version:
+          - 2.7
+          - 3.7
+        exclude:
+          - runner-os: ubuntu-latest
+            runner-python-version: 2.7
+
+    steps:
+      - name: Check out ${{ github.repository }} on disk
+        uses: actions/checkout@master
+      - name: Set up Python ${{ matrix.runner-python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.runner-python-version }}
+      - name: Set up pip cache
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('tests/sanity/requirements.txt') }}-${{ hashFiles('tests/unit/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+            ${{ runner.os }}-
+      - name: Install Ansible ${{ matrix.ansible-version }}
+        run: >-
+          python -m pip install --user ${{ matrix.ansible-version }}
+      - name: Build a collection tarball
+        run: >-
+          ~/.local/bin/ansible-galaxy collection build --output-path
+          "${GITHUB_WORKSPACE}/.cache/collection-tarballs"
+      - name: Store migrated collection artifacts
+        uses: actions/upload-artifact@v1
+        with:
+          name: collection
```

Comment:
@sshnaidm

Because this job is a part of the matrix, you need to have different names for artifacts or they'll probably overwrite each other. So there's no guarantee which of the jobs produces the artifact that the next one will download.

Also, I'm not sure if it makes sense to build all of the permutations. Maybe it's okay to build just one but test it under different envs? After all, you'll upload only one tarball to Galaxy and it's what the end-users will receive, not all possible variants.

Reply:
Agree, I'll make only one job building, it shouldn't matter which platform is used.